org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -3367,7 +3377,7 @@ public static List<String> getAllVersions() {
 	 * @see #getFirstJavaSourceVersionSupportedByCompiler()
 	 * @since 3.39
 	 */
-	public static List<String> getAllJavaSourceVersionsSupportedByCompiler() {
+	public static SortedSet<String> getAllJavaSourceVersionsSupportedByCompiler() {
```

Comment:
THis, if merged, would break JDT Debug build.
Please provide PR to workaround this, like
```
diff --git a/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/JREsPreferencePage.java b/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/JREsPreferencePage.java
index fbeced8..898e5b5 100644
--- a/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/JREsPreferencePage.java
+++ b/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/JREsPreferencePage.java
@@ -17,3 +17,4 @@
 import java.io.File;
-import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
@@ -200,4 +201,4 @@
 		else {
-			List<String> allVersions = JavaCore.getAllJavaSourceVersionsSupportedByCompiler();
-			String latest = allVersions.get(allVersions.size() - 1);
+			SortedSet<String> allVersions = new TreeSet<>(JavaCore.getAllJavaSourceVersionsSupportedByCompiler());
+			String latest = allVersions.last();
 			String vmver = null;
diff --git a/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/StandardVMPage.java b/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/StandardVMPage.java
index 14b82a6..acbe8f4 100644
--- a/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/StandardVMPage.java
+++ b/org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/StandardVMPage.java
@@ -18,3 +18,4 @@
 import java.net.URL;
-import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
@@ -249,4 +250,4 @@
 		if (s.isOK()) {
-			List<String> allVersions = JavaCore.getAllJavaSourceVersionsSupportedByCompiler();
-			String latest = allVersions.get(allVersions.size() - 1);
+			SortedSet<String> allVersions = new TreeSet<>(JavaCore.getAllJavaSourceVersionsSupportedByCompiler());
+			String latest = allVersions.last();
 			IVMInstallType vmType = fVM.getVMInstallType();
```
After that we could merge this PR.

Reply:
Done in https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/468.